runtime/src/message_processor.rs

```diff
@@ -34,6 +34,13 @@ impl ::solana_frozen_abi::abi_example::AbiExample for MessageProcessor {
     }
 }
 
+/// Resultant information gathered from calling process_message()
+#[derive(Debug, Default, Clone, Copy, PartialEq, Eq)]
+pub struct ProcessedMessageInfo {
+    /// The amount that the accounts data len has changed
+    pub accounts_data_len_delta: i64,
+}
```

Comment:
I was thinking that if accounts can (1) be closed, and (2) be reallocated, that it would be possible for the accounts data len delta to be negative too. So I've made this an `i64`. Is this line of thinking correct? Because if not, it would make it easier if the accounts data len can only grow.

Reply:
As an update, I've got a branch where I'm tracking the accounts data on _close_ from `bpf_loader::process_loader_upgradeable_instruction()`. I'm not sure if that's the right place to track account closure, but it was at least one place where I identified that accounts_data_len could be negative.

Reply:
Yeah, the builtin programs are a place where we should update the alloc meter.  Accounts will soon be reallocable so add/sub are both required of the meter.